src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -204,29 +207,79 @@ private static bool TryParseStatusLine(HttpResponseMessage response, string resp
                 return false;
             }
 
+            // Clear the header if status line is recieved again. This signifies that there are multiple response headers (like in redirection).
+            response.Headers.Clear();
+            
+            response.Content.Headers.Clear();        
+
+            int responseHeaderLength = responseHeader.Length;
+
             // Check if line begins with HTTP/1.1 or HTTP/1.0
             int prefixLength = s_httpPrefix.Length;
-            if ((responseHeader[prefixLength] == '1') && (responseHeader[prefixLength + 1] == '.'))
+            int versionIndex = prefixLength + 2;
+
+            if ((versionIndex < responseHeaderLength) && (responseHeader[prefixLength] == '1') && (responseHeader[prefixLength + 1] == '.'))
             {
-                if (responseHeader[prefixLength + 2] == '1')
+                if (responseHeader[versionIndex] == '1')
                 {
-                    response.Version = new Version(1, 1);
+                    response.Version = HttpVersion.Version11;
                 }
-                else if (responseHeader[prefixLength + 2] == '0')
+                else if (responseHeader[versionIndex] == '0')
                 {
-                    response.Version = new Version(1, 0);
+                    response.Version = HttpVersion.Version10;
```

Comment:
The WinHttpHandler class currently does `response.Version = null;` when it is unexpected.  However, our .NETNative Http.dll currently does `response.Version = new Version(0,0);`.

We need to standardize on one of those.  Our current thinking is that `null` is very unexpected and so we will probably reconcile to `new Version(0,0)'.